TESTS/network/wifi/README.md

```diff
@@ -0,0 +1,506 @@
+| 11  | WIFI_CONNECT                            |                            | MUST     |
+| 12  | WIFI_CONNECT_SECURE                     | With security type:        |          |
+|     |                                         | NSAPI_SECURITY_WEP         | SHOULD   |
+|     |                                         | NSAPI_SECURITY_WPA         | SHOULD   |
+|     |                                         | NSAPI_SECURITY_WPA2        | SHOULD   |
+|     |                                         | NSAPI_SECURITY_WPA_WPA2    | MUST     |
+| 13  | WIFI_CONNECT_SECURE_FAIL                |                            | MUST     |
+| 14  | WIFI_CONNECT_DISCONNECT_REPEAT          |                            | MUST     |
+| 15  | WIFI_SCAN_NULL                          |                            | SHOULD   |
+| 16  | WIFI_SCAN                               |                            | SHOULD   |
+
+Building test binaries
+----------------------
+
+For testing the board and driver, test against the Mbed OS
+master branch to get the most recent, up-to-date test cases and drivers.
+
+To create a build environment:
+
+```.sh
+mbed new wifi_test
+cd wifi_test
+cd mbed-os
+git checkout master
+cd ..
+```
+
+Now build test binaries:
+
+```.sh
+mbed test --compile -t <toolchain> -m <target> -n mbed-os-tests-network-wifi
+```
+
+Running tests
+-------------
+
+Bear in mind that RAAS user and password have to be set in the shell to access it.
+
+```.sh
+mbedgt -g <target>:raas_client:ruka.mbedcloudtesting.com:8000 -n mbed-os-tests-network-wifi -V -v
+```
+
+Test cases for WifiInterface class
+-------------------------------------
+
+### WIFI_CONSTRUCTOR
+
+**Description:**
+
+�Test that constructor of the driver works.
+
+**Preconditions:**
+
+1.  None
+
+**Test steps:**
+
+1.  Construct the driver by calling the constructor.
+
+**Expected result:**
+
+Constructor returns.
+
+### WIFI_CONNECT_NOCREDENTIALS
+
+**Description:**
+
+�Test `WiFiInterface::connect()` without parameters. Don't set parameters with `set_credentials()`
+
+This must be a first test to run after constructing the driver. No credentials should be given for the driver before this test.
+
+**Preconditions:**
+
+1.  Test enviroment is set up as specified in "Test Environment" chapter.
+
+**Test steps:**
+
+1.  Initialise the driver
+2.  Call�`WiFiInterface::connect()`
+3.  `disconnect()`
+
+**Expected result:**
+
+`connect()` call returns�NSAPI_ERROR_PARAMETER or�NSAPI_ERROR_NO_SSID
+
+### WIFI_SET_CREDENTIAL
+
+**Description:**
+
+This test case is to test whether the driver accepts valid credentials and reject ones that are not valid.
+
+**Preconditions:**
+
+1.  Driver class is initialised.
+
+**Test steps:**
+
+1.  Call `set_credentials()` with various parameters described in "Expected results"
+
+**Expected result:**
+
+| Parameters |                                |                                  | Expected return code                                |
```

Comment:
Can we change this table to
|ssid|password|security| Expected return code|
|----|---------|--------|-----------------------|

The we can remove those `ssid=` and `pass=` which fill up the table and make it harder to read.